huaweicloud/services/cce/data_source_huaweicloud_cce_clusters.go

```diff
@@ -0,0 +1,334 @@
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"client_certificate_data": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"client_key_data": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceCCEClustersV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	cceClient, err := config.CceV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to create HuaweiCloud CCE client : %s", err)
+	}
+
+	listOpts := clusters.ListOpts{
+		ID:                  d.Get("cluster_id").(string),
+		Name:                d.Get("name").(string),
+		Type:                d.Get("cluster_type").(string),
+		Phase:               d.Get("status").(string),
+		VpcID:               d.Get("vpc_id").(string),
+		EnterpriseProjectID: config.GetEnterpriseProjectID(d),
+	}
+
+	refinedClusters, err := clusters.List(cceClient, listOpts)
+	logp.Printf("[DEBUG] Value of allClusters: %#v", refinedClusters)
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to retrieve clusters: %s", err)
+	}
+
+	ids := make([]string, 0)
+	clustersToSet := make([]map[string]interface{}, 0, len(refinedClusters))
+
+	for _, v := range refinedClusters {
+
+		ids = append(ids, v.Metadata.Id)
+
+		cluster := map[string]interface{}{
+			"name":                   v.Metadata.Name,
+			"id":                     v.Metadata.Id,
+			"status":                 v.Status.Phase,
+			"flavor_id":              v.Spec.Flavor,
+			"cluster_version":        v.Spec.Version,
+			"cluster_type":           v.Spec.Type,
+			"description":            v.Spec.Description,
+			"billing_mode":           v.Spec.BillingMode,
+			"vpc_id":                 v.Spec.HostNetwork.VpcId,
+			"subnet_id":              v.Spec.HostNetwork.SubnetId,
+			"container_network_cidr": v.Spec.ContainerNetwork.Cidr,
+			"container_network_type": v.Spec.ContainerNetwork.Mode,
+			"eni_subnet_id":          v.Spec.EniNetwork.SubnetId,
+			"eni_subnet_cidr":        v.Spec.EniNetwork.Cidr,
+			"authentication_mode":    v.Spec.Authentication.Mode,
+			"security_group_id":      v.Spec.HostNetwork.SecurityGroup,
+			"enterprise_project_id":  v.Spec.ExtendParam["enterpriseProjectId"],
+			"service_network_cidr":   v.Spec.KubernetesSvcIPRange,
+		}
+
+		var endpoints []map[string]interface{}
+		for _, endpoint := range v.Status.Endpoints {
+
+			mapping := map[string]interface{}{
+				"url":  endpoint.Url,
+				"type": endpoint.Type,
+			}
+			endpoints = append(endpoints, mapping)
+		}
+		cluster["endpoints"] = endpoints
+
+		r := clusters.GetCert(cceClient, v.Metadata.Id)
+
+		kubeConfigRaw, err := utils.JsonMarshal(r.Body)
+
+		if err != nil {
+			logp.Printf("Error marshaling r.Body: %s", err)
+		}
+
+		cluster["kube_config_raw"] = string(kubeConfigRaw)
```

Comment:
is it necessary to add `kube_config_raw` as it is too huge?

Reply:
It's a useful attribute. I prefer to keep it. And we can consider a better way to save this attribute.